src/app/shared/components/editable-form/editable-form.component.html

```diff
@@ -0,0 +1,47 @@
+@if (isEditMode) {
+  <form class="wrapper" [formGroup]="form()" (ngSubmit)="save()">
+    <div class="inputs">
+      @for (controlName of form().controls | keyvalue; track $index) {
+        @switch (typeInputs()) {
+          @case ('text') {
+            <tui-input class="input" [formControlName]="controlName.key">
+              {{ controlName.key | titlecase }}
+              <input [placeholder]="controlName.key | titlecase" tuiTextfieldLegacy />
+            </tui-input>
+          }
+          @case ('date-time') {
+            <tui-input-date-time class="input" [formControlName]="controlName.key" [min]="minDate">
+              {{ controlName.key | titlecase }}</tui-input-date-time
+            >
+          }
+          @case ('price') {
+            <tui-input-number
+              class="input"
+              [formControlName]="controlName.key"
+              [tuiTextfieldPrefix]="'USD' | tuiCurrency"
```

Comment:
Magic string?

Reply:
then all the strings that are in the template are magic?

Reply:
Currency could change in the future, it should be variable in any case